main.py

```diff
@@ -29,25 +30,38 @@ def process_email(email_path):
         elif k.lower() == "n":
             logging.info("---------Stop processing emails---------")
             return
+    email_counter = 0  
     for mail in mail_info:
         state, data = chatbot.get_content(mail)
         if state == 'Succeed':
             res.append(data)
-    return res
+            email_counter += 1
+            if email_counter % AUTO_SAVE_EMAIL == 0:
+                export_to_csv(res, output_csv)
+                res = []
+    return res, email_counter
 
 def export_to_csv(data, filename):
-    with open(filename, mode='w', newline='', encoding='utf-8') as csvfile:
+    file_exists = False
+    try:
+        with open(filename, 'r', newline='', encoding='utf-8') as csvfile:
+            file_exists = True
+    except FileNotFoundError:
+        pass
```

Comment:
Do not use `pass` when you see `FileNotFoundError`, we can use mode = 'a' no matter whether the file exists or not.

Reply:
We need check if file exist to write header